### DISCUSSION TITLE
Representing a graph or tree in the TUI

### DISCUSSION THREAD

**Discussion by mickael-menu** [zk-org/zk#125] [Ideas]
`zk graph` returns a JSON representation of the link graph, but it would be interesting to have a more visual output as well.

Printing a full-blown graph would be difficult, [there are text solutions](https://github.com/ggerganov/dot-to-ascii) but that would only work with a very limited set of notes.

A more suitable alternative could be to simplify the graph into a tree. We can take inspiration from [`cargo tree`](https://doc.rust-lang.org/cargo/commands/cargo-tree.html), e.g.

```
myproject v0.1.0 (/myproject)
└── rand v0.7.3
    ├── getrandom v0.1.14
    │   ├── cfg-if v0.1.10
    │   └── libc v0.2.68
    ├── libc v0.2.68 (*)
    ├── rand_chacha v0.2.2
    │   ├── ppv-lite86 v0.2.6
    │   └── rand_core v0.5.1
    │       └── getrandom v0.1.14 (*)
    └── rand_core v0.5.1 (*)
```

Items suffixed with `(*)` have been "de-duplicated" by `cargo tree`.

Here's an example with more Zettelkasten-y items:

```
    §How to invest in the stock markets?
    │
    ├── Financial markets are random
    │   ├── Stick to your portfolio strategy
    │   │   └── Couch potato investment strategy
    │   │
    │   └── Compound interests make you rich
    │
    ├── Compound interests make you rich
    │
    └── Investment business is a scam
```

**Comment by prmadev** [comment]
This seems to be an interesting idea, however, I think the tree-like structure is only possible, if the notes follow a stricktly tree-like structure. Which I think even with neurons version of folgezettel might not be possible considering that each node, might have more than one parent. 
Unless if the parent tree is sepereted from the children tree.

Another thing to consider is that one might just use one note and it's context, instead of the whole zettelkasten. This way, assuming small number of links, the problem of big number of notes can be mitigated.

**Reply by mickael-menu** [reply]
Yes it would probably be usable only with a single root note, like with `cargo tree`.

**Comment by prncss-xyz** [comment]
Opened a feature request in a [place](https://github.com/prncss-xyz/neo-tree-zk.nvim/issues/2) where it can be done.
